### PR TITLE
unique constraint violation

### PR DESCRIPTION
**Pull request by kashishchakarvarti**

### CODE REVIEW

**Comment by nawazdhandala** [comment]
@kashishchakarvarti build fails.

**Comment by kashishchakarvarti** [comment]
will take a look

File: services/cloudObjects.js [hunk 1]
```diff
@@ -496,18 +496,22 @@ var _isSchemaValid = function(appId, collectionName, document, accessList, isMas
                     query.$or.push(temp);
                     console.log('OR Query');
                     console.log(query);
+                    uniqueColumn = temp
```
Comment (nawazdhandala) [inline services/cloudObjects.js]: I dont understand this line. What's this about?

File: services/cloudObjects.js [hunk 2]
```diff
@@ -496,18 +496,22 @@ var _isSchemaValid = function(appId, collectionName, document, accessList, isMas
...
-                promises.push(findPromise.promise);
+                    promises.push(findPromise.promise);
                 global.mongoService.document.find(appId, collectionName, query, null, null, 9999999, 0, null, true).then(function(res) {
                     console.log('Unique Result');
                     console.log(res);
                     if (res.length === 1 && res[0]._id === document._id) {
                         findPromise.resolve('Update the document');
-                    } else if (res.length > 0) {
-                        findPromise.reject('Unique constraint violated.');
+                    } else if (res.length > 0) { console.log(uniqueColumn)
```
Comment (nawazdhandala) [inline services/cloudObjects.js]: Remove console log.

File: services/cloudObjects.js [hunk 3]
```diff
@@ -496,18 +496,22 @@ var _isSchemaValid = function(appId, collectionName, document, accessList, isMas
...
+                        key = Object.keys(uniqueColumn);
+                        var violatedColumn = key[0]              
+                       findPromise.reject({error:'Unique constraint violated',violatedColumn:violatedColumn});
```
Comment (nawazdhandala) [inline services/cloudObjects.js]: I'm pretty sure you need to reject the mainPromise and not the findPromise.

**Pull request closed without merging**